homeassistant/components/bangolufsen/__init__.py

```diff
@@ -0,0 +1,83 @@
+"""The Bang & Olufsen integration."""
+from __future__ import annotations
+
+import logging
+from multiprocessing.pool import ApplyResult
+from typing import cast
+
+from mozart_api.models import BatteryState
+from mozart_api.mozart_client import MozartClient
+from urllib3.exceptions import MaxRetryError
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_NAME, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError, ConfigEntryNotReady
+from homeassistant.helpers.event import async_call_later
+
+from .const import DOMAIN, ENTITY_ENUM, WEBSOCKET_CONNECTION_DELAY
+from .media_player import BangOlufsenMediaPlayer
+from .websocket import BangOlufsenWebsocket
+
+PLATFORMS = [Platform.MEDIA_PLAYER]
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    # Ensure that a unique id is available
+    if not entry.unique_id:
+        raise ConfigEntryError("Can't retrieve unique id from config entry. Aborting")
+
+    # If connection can't be made abort.
```

Comment:
Why do we abort here? Since this is a media player I would expect it to be a very common case that the media player is powered off, it seems reasonable that the media player entity is still created in that case, but unavailable?

Reply:
Yes, this works as expected.